app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -1470,6 +1472,30 @@ class PluginApi {
       { ignoreMissing: true }
     );
   }
+
+  pmTopicTrackingStateInterface() {
+    const pmTopicTrackingState = this._lookupContainer(
+      "pm-topic-tracking-state:main"
+    );
+
+    return EmberObject.extend({
```

Comment:
I'm curious why you extend an object to do this rather than call create with these methods?

Reply:
I was just following the Ember guide [here](https://guides.emberjs.com/v3.4.0/object-model/classes-and-instances/#toc_defining-classes) to define a new class. Do you have an example of how I can pass the functions in when calling `create`?

Reply:
It's true that `extend` creates a class, but you are only ever making one instance of that class so it's not very useful. You can pass functions in when calling `create`

```js
let x = EmberObject.create({
  hello() {
     console.log('world');
  }
});
x.hello();  // works
```